src/OpenFeature/Error/FlagNotFoundException.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using OpenFeature.Constant;
+
+namespace OpenFeature.Error
+{
+    /// <summary>
+    /// Provider was unable to find the flag error when evaluating a flag.
+    /// This exception should be thrown when evaluating a flag inside a IFeatureFlag provider
```

Comment:
```suggestion
```

Reply:
How would people feel about something aside from "should"? Should makes it sound like you should throw these exceptions, when really you can/could, but I wouldn't personally encourage it. The SDK also supports returning error codes, and the performance (and application debug-ability)  of that will be better than an exception. (Which probably matters more to FlagNotFound, evaluating a flag that doesn't exist yet, or has been deleted, isn't an exceptional situation. Versus an invalid context, which is more of a programming error, and therefore less performance critical.)

Reply:
removed. Thanks for the comments. I leave this sentence because it was mentioned in FeatureProviderException. And all these exceptions derive from that base (having this commented line here).

Reply:
@valentk777 has removed the line entirely, which I think is the right move. I agree the "should" seemed like more of a directive than we'd want.